xbmc/interfaces/AnnouncementManager.cpp

```diff
@@ -30,6 +30,203 @@ using namespace KODI;
 
 const std::string CAnnouncementManager::ANNOUNCEMENT_SENDER = "xbmc";
 
+namespace
+{
+
+void CopyPVRTagInfoToObject(const PVR::CPVRChannel& channel, bool copyPlayerId, CVariant& object)
+{
+  auto& objItem = object["item"];
+
+  objItem["type"] = "channel";
+  objItem["title"] = channel.ChannelName();
+  objItem["channeltype"] = channel.IsRadio() ? "radio" : "tv";
+
+  if (copyPlayerId)
+  {
+    object["player"]["playerid"] =
+        static_cast<int>(channel.IsRadio() ? PLAYLIST::Id::TYPE_MUSIC : PLAYLIST::Id::TYPE_VIDEO);
+  }
+
+  objItem["id"] = channel.ChannelID();
+}
+
+void CopyVideoTagInfoToObject(CFileItem& item, CVariant& object)
+{
+  CVideoInfoTag& tag = *item.GetVideoInfoTag();
+
+  auto& objItem = object["item"];
+  int id = tag.GetDatabaseId();
+
+  //! @todo Can be removed once this is properly handled when starting playback of a file
+  if (id <= 0 && !item.GetPath().empty() && item.GetProperty(LOOKUP_PROPERTY).asBoolean(true))
+  {
+    CVideoDatabase videodatabase;
+    if (videodatabase.Open())
+    {
+      std::string videoInfoTagPath = tag.m_strFileNameAndPath;
+      std::string path;
+      if (StringUtils::StartsWith(videoInfoTagPath, "removable://"))
+        path = videoInfoTagPath;
+      else
+        path = item.GetPath();
+      if (videodatabase.LoadVideoInfo(path, tag, VideoDbDetailsNone))
+        id = tag.GetDatabaseId();
+
+      videodatabase.Close();
+    }
+    else
+    {
+      CLog::Log(LOGWARNING, LOGANNOUNCE,
+                "{} - Unable to open video database. Can not load video tag for announcement!",
+                __FUNCTION__);
```

Comment:
Instead of using `__FUNCTION__`, use `CLog::LogFC` - The F takes care to log method signature, the C stands for the log component, which is `LOGANNOUNCE` here.